misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -164,23 +170,59 @@ private List<Path> getDependenciesForJacocoXML(JBallerinaBackend jBallerinaBacke
     /**
      * Generate the json coverage report for Testerina.
      *
-     * @param orgName package org name
-     * @param packageName package name
+     * @param orgName          package org name
+     * @param packageName      package name
      * @param filteredPathList List of the extracted source path
      * @return CoverageBuilder
      * @throws IOException
      */
     private CoverageBuilder generateTesterinaCoverageReport(String orgName, String packageName,
-                                                            List<Path> filteredPathList) throws IOException {
+                                                            List<Path> filteredPathList,
+                                                            Map<String, Module> originalModules) throws IOException {
         // For the Testerina report only the ballerina specific sources need to be extracted
         addCompiledSources(filteredPathList, orgName, packageName);
         execFileLoader.load(executionDataFile.toFile());
         final CoverageBuilder coverageBuilder = analyzeStructure();
-        // Create Testerina coverage report
-        createReport(coverageBuilder.getBundle(title), moduleCoverageMap);
+
+        // Analyze original modules
+        List<DocumentId> excludedFiles = getGeneratedFilesToExclude(coverageBuilder.getSourceFiles(), originalModules);
+
+        // If there are files to exclude, we add that to the exclusion list
+        if (excludedFiles.isEmpty() == true) {
+            createReport(coverageBuilder.getBundle(title), moduleCoverageMap);
+        } else {
+            createReport(coverageBuilder.getBundle(title), moduleCoverageMap, excludedFiles);
+        }
+
+        filterGeneratedCoverage(coverageBuilder.getBundle(title), moduleCoverageMap, originalModules);
         return coverageBuilder;
     }
 
+    private List<DocumentId> getGeneratedFilesToExclude(Collection<ISourceFileCoverage> sourceFileCoverages,
+                                                        Map<String, Module> originalModules) {
+        List<DocumentId> excludedDocumentIds = new ArrayList<>(this.module.documentIds());
```

Comment:
We can simplify this method further as follows:
```java
List<DocumentId> originalDocumentIdsList = new ArrayList<>(originalModule.documentIds());
List<DocumentId> modifiedDocumentIdsList = new ArrayList<>(this.module.documentIds());

modifiedDocumentIdsList.removeAll(originalDocumentIdsList); // This will retain the generated files and remove the rest
```

Reply:
Changed to make use of removeAll